From f6bddc1870a820745d7dc81a142bad9f368c9aa6 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Wed, 10 Sep 2014 22:01:06 +1000
Subject: [PATCH 2/4] AVFD: Do not force the number of channels for AC3
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Honestly, I can’t fully recall what it was ever there for. I remember something to do with properly detecting channel changes, but it doesn’t appear to make a difference anymore.
---
 mythtv/libs/libmythtv/avformatdecoder.cpp |    2 --
 1 file changed, 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index 873f001..d6701ee 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -4688,8 +4688,6 @@ bool AvFormatDecoder::ProcessAudioPacket(AVStream *curstream, AVPacket *pkt,
                 {
                     ctx->request_channel_layout =
                         av_get_default_channel_layout(m_audio->GetMaxChannels());
-                    if (ctx->codec_id == AV_CODEC_ID_AC3)
-                        ctx->channels = m_audio->GetMaxChannels();
                 }
                 else
                 {
-- 
1.7.10.2

